source/common/modules/markdown-editor/hooks/codeblock-classes.js

```diff
@@ -45,6 +47,14 @@ function applyCodeblockClasses (cm) {
     const wrapClass = (info.wrapClass !== undefined) ? String(info.wrapClass) : ''
     const isCurrentlyCode = wrapClass.includes(codeblockClass)
 
+    if (i > 0 && codeBlockRE.test(cm.lineInfo(i - 1).text)) {
```

Comment:
Potentially faster solution (especially useful for documents with a lot of code blocks): `cm.getLine(i - 1)`. This doesn't include all the metadata of the line which we don't need here. (and in the next if-condition)

Reply:
Additional idea: We can also leverage the variable `isCodeBlock`. If that variable switches from false to true, this means the current line is the first line of a code block. If the variable switches from true to false, this means the previous line is the last line of a code block. If the variable remains `true`, this indicates it's regular code block line.

This might make this piece of code many times faster because we can refrain from any additional regular expression checks.